tests/Opus/JobTest.php

```diff
@@ -68,12 +69,14 @@ public function testEqualJobsHaveEqualHashes()
         $job1 = new Job();
         $job1->setLabel('JobTest');
         $job1->setData('somedata');
+        $sha1Id1 = $job1->generateSha1Id();
```

Comment:
Okay, ich verstehe jetzt, das sich das Verhalten von `getSha1Id` geändert hat. Das geht aber nicht ohne weiteren Hinweis. Ich würde hoffen, das `getSha1Id` nicht für noch nicht gespeicherte Jobs aufgerufen wird, ich wüsste nicht warum, aber trotzdem ist es eine Änderung im Verhalten.

Die Tests in denen `getSha1Id` durch `generateSha1Id` ersetzt wurde, sollte zurück geändert werden. Ein Aufruf von GET kann die ID bei Bedarf durch einen Aufruf von GENERATE erzeugen.

Ein weiteres Problem ergibt sich dadurch, dass Doctrine die GET Funktion nicht verwendet, sondern direkt auf die Variablen zugreift. Das wird aber durch die `preStore`-Funktion gelöst. Das kann erst einmal so bleiben. Doctrine selbst hat aber vermutlich ein Interface für solche Funktionen.

Reply:
Ich glaube, das Verhalten von `getSha1Id()` hat sich nicht geändert. Denn `getSha1Id()` war früher (auch wenn es so scheint) _nicht_ der Getter für die `sha1_id` Spalte in der `jobs` Datenbank-Tabelle, sondern die Methode zur Generierung der SHA1 ID. Dies ist auch weiterhin so, nur dass ich diese Methode in `generateSha1Id()` umbenannt habe. Damit kann `getSha1Id()` jetzt ganz regulär als Getter verwendet werden. Das Ganze sollte also eigentlich nur eine Umbenennung sein und keine Änderung in der Funktion der Tests.